flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -366,11 +366,10 @@ private void registerAsyncCheckpointRunnable(long checkpointId, AsyncCheckpointR
 		synchronized (lock) {
 			if (closed) {
 				LOG.debug("Cannot register Closeable, this subtaskCheckpointCoordinator is already closed. Closing argument.");
-				final boolean running = asyncCheckpointRunnable.isRunning();
 				closeQuietly(asyncCheckpointRunnable);
 				checkState(
-					!running,
```

Comment:
In the past we might encounter unnecessary exception while `#registerAsyncCheckpointRunnable` and `AsyncCheckpointRunnable#close` execute concurrently. Now we only close the runnable quietly without throwing any exception for `closed` case.

I agree with this fix, but do we have any existing ITCase for covering/verifying this change?

Reply:
I think there are no ITCases covering this, but some existing end-to-end tests indirectly cover this.

Reply:
I am a bit torn whether it deserves somehow UT or ITCase for the changes.
It seems not bad to be covered by existing E2E tests, so I am fine with you final judgement here.

Reply:
Then let's leave it as-is for now.